### PR TITLE
Feature/refactor traefik package

### PR DESCRIPTION
**Pull request by avnik**
Move confifuration stuff to own package

### CODE REVIEW

**Comment by ryane** [comment]
lgtm but I don't have a big history with the traefik package. @BrianHicks look good to you?

File: packages/traefik/spec.yml [hunk 1]
```diff
@@ -32,55 +31,26 @@ targets:
...
-        useradd -r \
-                -g {{.Name}} \
-                -d /etc/{{.Name}} \
-                -s /sbin/nologin \
-                -c "{{.Name}} user" \
-                {{.Name}}
-
   after-install: |
-    chown -R {{.Name}}:{{.Name}} /etc/{{.Name}}
+    chown -R root:root /etc/{{.Name}}
```
Comment (BrianHicks) [inline packages/traefik/spec.yml]: why root instead of `traefik`?

File: packages/traefik-config-scripts/spec.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+---
```
Comment (BrianHicks) [inline packages/traefik-config-scripts/spec.yml]: yes yes yes yes yes

**Comment by BrianHicks** [comment]
@ryane added notes inline. Mostly looks good, just a question for @avnik :)

File: packages/traefik/spec.yml (hunk 1 above)
Comment (avnik) [inline packages/traefik/spec.yml]: @BrianHicks  Should traefik _write_ to own configs? or they written by root, and remain immutable?

PS But good notice here to make something like root:traefik, and 0750  if  config contain credentionals

File: packages/traefik-config-scripts/spec.yml (hunk 2 above)
Comment (avnik) [inline packages/traefik-config-scripts/spec.yml]: actually spec.yml from traefik have same "signature" in begin.
Actually I copy it, and remove unrelated as well as remove unrelated form original spec

Comment (BrianHicks) [inline packages/traefik-config-scripts/spec.yml]: haha no, I'm just happy to see that you've separated the two packages. No change needed. :)

File: packages/traefik/spec.yml (hunk 1 above)
Comment (BrianHicks) [inline packages/traefik/spec.yml]: As long as the traefik user can read them, we're good. *BUT* it looks like you've also removed the traefik user. How's that work? The service is still run as the traefik user, is it not?

Comment (avnik) [inline packages/traefik/spec.yml]: @BrianHicks Is here -- https://github.com/asteris-llc/mantl-packaging/pull/134/files/95fd6cf114b34d672bc1d0cf56b4d41810ea409e#diff-189888afe84055bbd5a83450b1a6be8dR43 just moved to config package, this part should be shared with cisco stuff

Comment (BrianHicks) [inline packages/traefik/spec.yml]: The user probably needs to be in the traefik package. We're going to eventually want to be able to completely separate the config scripts from the binary—they're less helpful than I had hoped.

File: packages/traefik/spec.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
 name: traefik
 version: 1.0.0
-iteration: 1
+iteration: 3
```
Comment (BrianHicks) [inline packages/traefik/spec.yml]: does this have to change to 3 or can it be 2?

**Comment by BrianHicks** [comment]
other than the question about the the iteration number, LGTM as long as it installs successfully :)

**Comment by avnik** [comment]
@BrianHicks I just felt than set same iteration like main traefik would be good idea. Let merge/upload then ;)

**Pull request merged**